### PR TITLE
[Merged by Bors] - CI: extract update comment and merge import and diff summaries

### PR DESCRIPTION
**Pull request by adomani**
This PR extends the PR summary introduced in #13556.

* Extracts the GitHub comment creation/update to a new file;
* adds the output of `move-decls` to the automatically updated message posted by the import-automation;
* if the output of either summary exceeds a few lines, then that output is hidden behind a collapsed summary.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: scripts/update_PR_comment.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+  # the text of the message that will replace the current one
+  message="${1}" #=$(python ./scripts/import-graph-report.py base.json head.json changed_files.txt)
```
Comment (jcommelin) [inline scripts/update_PR_comment.sh:4]: I guess this comment can now be removed?
```suggestion
  message="${1}"
```

File: .github/workflows/import_count.yml [hunk 2]
```diff
@@ -34,33 +34,36 @@ jobs:
...
-            curl -s -S -H "Content-Type: application/json" -H "$headers" -X "$method" -d "$data" "$url"
-        fi
+        PR="${{ github.event.pull_request.number }}"
+        title="### PR summary"
+
+        ## Import count comment
+        importCount=$(python ./scripts/import-graph-report.py base.json head.json changed_files.txt)
+
+        ## Declarations' diff comment
+        declDiff=$(./scripts/no_lost_declarations.sh short)
```
Comment (jcommelin) [inline .github/workflows/import_count.yml:65]: So this workflow is now responsible for both the import graph report and the move-decls bit.

I think that makes sense. But maybe we should rename the workflow file.

Comment (adomani) [reply .github/workflows/import_count.yml:65]: Thanks!  I renamed it `PR_summary.yml`.

File: .github/workflows/PR_summary.yml [hunk 3]
```diff
@@ -34,33 +34,36 @@ jobs:
...
-        git checkout origin/master scripts/count-trans-deps.py scripts/import-graph-report.py
+        # ***switch*** to `origin/master`, when merging or right after
+        git checkout origin/adomani/move_decls_autocomment scripts/update_PR_comment.sh
 
         # Compute the counts for the merge base
         python ./scripts/count-trans-deps.py "Mathlib/" > base.json
 
+        # switch back to the current branch: the `no_lost_declarations` script should be here
+        git checkout "${currentHash}"
     - name: Compare results and comment
```
Comment (jcommelin) [inline .github/workflows/PR_summary.yml:54]:
```suggestion
    - name: Post or update the summary comment
```

File: scripts/no_lost_declarations.sh [hunk 4]
```diff
@@ -104,25 +104,28 @@ else
   grep '\(+\|-\)' | sed 's=^ *1 =`=; s=^[^`]=`=; s=$=`='
 fi)"
 
+printf '*Diff of declarations*\n\n'
+
```
Comment (jcommelin) [inline scripts/no_lost_declarations.sh:108]: Should the entire report also be inside a `<details>` if it is longer than X lines? (X = 5?)

File: .github/workflows/PR_summary.yml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-name: Compute transitive imports
+name: Compute PR summary
```
Comment (jcommelin) [inline .github/workflows/PR_summary.yml:1]:
```suggestion
name: Post PR summary comment
```

File: .github/workflows/PR_summary.yml [hunk 6]
```diff
@@ -34,33 +34,36 @@ jobs:
...
 
         # Compute the counts for the HEAD of the PR
         python ./scripts/count-trans-deps.py "Mathlib/" > head.json
 
         # Checkout the merge base
         git checkout "$(git merge-base master ${{ github.sha }})"
+        # git checkout origin/master scripts/count-trans-deps.py scripts/import-graph-report.py
         # get the files that we need from the current PR
-        git checkout origin/master scripts/count-trans-deps.py scripts/import-graph-report.py
+        # ***switch*** to `origin/master`, when merging or right after
```
Comment (jcommelin) [inline .github/workflows/PR_summary.yml:46]:
```suggestion
        # TODO ***change*** to `origin/master` in the line below, when merging or right after
```

**Review by jcommelin — COMMENTED** [review]
This seems to be converging.

File: scripts/import-graph-report.py [hunk 7]
```diff
@@ -51,7 +51,7 @@ def compare_counts(base_file, head_file, changed_files_txt):
         messages.append(f'| {file} | {base_count} | {head_count} | {sign}{diff} ({sign}{percent:.2f}%) |')
 
     # Build the message
-    message = '### Import summary\n\n'
+    message = ''
     if messages:
         message += '<details><summary>Dependency changes</summary>\n\n'
```
Comment (jcommelin) [inline scripts/import-graph-report.py:56]: Here the `<details>` should now be removed, no?

Comment (adomani) [reply scripts/import-graph-report.py]: Removed!

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**